### DISCUSSION TITLE
Should updating a non-existing property raise an exception?

### DISCUSSION THREAD

**Discussion by nerdoc** [django-commons/django-unicorn#354] [Ideas]
By creating non-existing Unicorn models and misusing the updated() method for saving their values, I came across the idea: why doesn't Unicorn raise an Exception if the frontend tries to update a value of a model=property that does not exist on the backend?
Given an input that has a `unicorn:model="email"` attr, and the following backend component:

```python
class EmailView(UnicornView):
    mail = "you@example.com"
    def updated(self, property_name, value):
       if property_name  == "mail":
           print("Somone changed the email value!")
```

... nothing happens in the backend, if you  change the input value by typing. It clearly contains a bug, notably a typo - in the fronend, `email` is used, the backend uses `mail` as variable name.
This is something where you can sit  for a long time and search for the error...

What yould you say if Unicorn would raise an Exception in `set_property_value`, at the end of the `if` tree instead of `break`?
Something like 
```python
raise KeyError(f"Property {property_name_part} was not found in the backend component. "
    "Make sure that you didn't misspell it.")
```

**Comment by adamghill** [comment]
I'm trying to think of a case where this would cause potential problems, but I can't off the top of my head. I do wonder if a warning would be a good first step to see if it causes any issues before throwing an actual exception? Any chance you want to throw together a PR for me to look at? :)

**Reply by nerdoc** [reply]
Sure, I'll try my best. warning is OK for me - a messages.error() should do it as well. No KeyError needed, you're right.
